Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/TranslationHandles.cs

```diff
@@ -0,0 +1,338 @@
+                    config.TranslationHandlePrefabColliderType, CursorContextInfo.CursorAction.Move, config.ColliderPadding, parent, config.DrawTetherWhenManipulating);
+
+                handles.Add(faceCenter.transform);
+            }
+
+            VisualUtils.HandleIgnoreCollider(config.HandlesIgnoreCollider, handles);
+
+            objectsChangedEvent.Invoke(this);
+        }
+
+        protected override void RecreateVisuals()
+        {
+            Debug.Log("RecreateVisuals");
+            for (int i = 0; i < handles.Count; ++i)
+            {
+                // get parent of visual
+                Transform obsoleteChild = handles[i].Find("visuals");
+                if (obsoleteChild)
+                {
+                    // get old child and remove it
+                    obsoleteChild.parent = null;
+                    Object.Destroy(obsoleteChild.gameObject);
+                }
+                else
+                {
+                    Debug.LogError("couldn't find translation visual on recreating visuals");
+                }
+
+                // create new visual
+                Bounds visualBounds = CreateVisual(i, handles[i].gameObject);
+
+                // update handle collider bounds
+                UpdateColliderBounds(handles[i], visualBounds.size);
+            }
+
+            objectsChangedEvent.Invoke(this);
+        }
+
+        protected override void UpdateColliderBounds(Transform handle, Vector3 visualSize)
+        {
+            Debug.Log("UpdateColliderBounds");
+            var invScale = visualSize.x == 0.0f ? 0.0f : config.HandleSize / visualSize.x;
+            GetVisual(handle).transform.localScale = new Vector3(invScale, invScale, invScale);
+            Vector3 colliderSizeScaled = visualSize * invScale;
+            if (config.TranslationHandlePrefabColliderType == HandlePrefabCollider.Box)
+            {
+                BoxCollider collider = handle.gameObject.GetComponent<BoxCollider>();
+                collider.size = colliderSizeScaled;
+                collider.size += BaseConfig.ColliderPadding;
+            }
+            else
+            {
+                SphereCollider collider = handle.gameObject.GetComponent<SphereCollider>();
+                collider.radius = colliderSizeScaled.x * 0.5f;
+                collider.radius += VisualUtils.GetMaxComponent(config.ColliderPadding);
+            }
+        }
+
+
+        protected Bounds CreateVisual(int handleIndex, GameObject parent)
+        {
+            GameObject midpointVisual;
+            GameObject prefabType = config.HandlePrefab;
+            if (prefabType != null)
+            {
+                midpointVisual = Object.Instantiate(prefabType);
+            }
+            else
+            {
+                midpointVisual = GameObject.CreatePrimitive(PrimitiveType.Sphere);
+                Object.Destroy(midpointVisual.GetComponent<SphereCollider>());
```

Comment:
What if TranslationHandlePrefabColliderType is of type Sphere, should we still destroy the SphereCollider in that case? Or if it's box, will the collider get created for this?

i.e. the part that handles collider recreates will only do if one isn't destroyed (lines 41 to 50ish)

Reply:
@thalbern , what is the situation with this? I am not very familiar with how you do the collider mechanisms. Would appreciate some eyeballs on this.

Reply:
we only destroy this now when we're not collider type sphere :)